src/NServiceBus.Unity/PropertyInjectionBuilderStrategy.cs

```diff
@@ -13,23 +12,15 @@ public PropertyInjectionBuilderStrategy(UnityObjectBuilder unityContainer)
             this.unityContainer = unityContainer;
         }
 
-        private IUnityContainer GetUnityFromBuildContext(IBuilderContext context)
-        {
-            var lifetime = context.Policies.Get<ILifetimePolicy>(NamedTypeBuildKey.Make<IUnityContainer>());
-            return lifetime.GetValue() as IUnityContainer;
-        }
-
         public override void PreBuildUp(IBuilderContext context)
         {
             if (mustInjectProperties)
             {
                 var type = context.BuildKey.Type;
                 var target = context.Existing;
-                var container = GetUnityFromBuildContext(context);
-
                 if (!type.FullName.StartsWith("Microsoft.Practices"))
                 {
-                    unityContainer.SetProperties(target.GetType(), target, container);
+                    unityContainer.SetProperties(target.GetType(), target, t => context.NewBuildUp(new NamedTypeBuildKey(t)));
```

Comment:
This switches us from using container.Resolve to context.NewBuildUp, so the context and resolution lifecycles are respected. This is a fix applied to 7.x +, but is needed here as well.